### ISSUE TITLE
When generating bindings for custom components, allow users to provide a denylist of symbols which should not be generated

### ISSUE THREAD

**Issue by ivmarkov** [esp-rs/esp-idf-sys#183]
See here for more details: https://github.com/esp-rs/esp-idf-hal/issues/215#issuecomment-1462363166

**Comment by N3xed** [comment]
This is probably not feasible, as, without the `bindings_module` parameter set, the same bindgen instance that generates the main `esp-idf` bindings will also generate those of the extra components. And to my knowledge, there is no way to exclude symbols of specific C headers.

**Comment by ivmarkov** [comment]
Yeah, but then having a global dentist would be just as good, as we are not trying to prevent collisions between the components and esp idf, but rather, between C code and Rust.
